packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -121,8 +170,82 @@ class ExpansionPanelList extends StatelessWidget {
   /// The duration of the expansion animation.
   final Duration animationDuration;
 
+  // Whether multiple panels can be open simultaneously
+  final bool _allowOnlyOnePanelOpen;
+
+  @override
+  State<StatefulWidget> createState() => new _ExpansionPanelListState();
+}
+
+class _ExpansionPanelListState extends State<ExpansionPanelList> {
+  ExpansionPanelRadio _currentOpenPanel;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget._allowOnlyOnePanelOpen) {
+      for (int i = 0; i < widget.children.length; i += 1) {
```

Comment:
Since you don't need the index `i` for anything, use `for (ExpansionPanelRadio child in widget.children)`, here and elsewhere.  I don't think you need to assert in `initState` unless you are dealing with the radio -> non radio or reverse